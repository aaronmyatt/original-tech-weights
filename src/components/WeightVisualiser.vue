<template>
<div class="weight-list progress" style="height: 20px;">
  <div v-for="weight in transformWeightList" :style="{ width: weight.weight, backgroundColor: weight.color }" class="weight progress-bar" role="progressbar" aria-valuenow="25" aria-valuemin="0" aria-valuemax="100">
    {{ weight.name }}: {{ weight.weight }}
  </div>
</div>
</template>

<script>
export default {
  name: 'WeightVisualiser',
  props: ['weightList'],
  computed: {
    transformWeightList() {
      const colors = ['green', 'blue', 'red', 'black', 'white'];
      return this.weightList.map((w) => {
        const wc = Object.assign({}, w);
        wc.weight = `${w.weight}%`;
        wc.color = colors.shift();
        return wc;
      });
    },
  },
};
</script>

<style scoped>
</style>
