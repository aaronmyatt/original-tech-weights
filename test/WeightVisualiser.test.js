import {
  shallow
} from 'vue-test-utils'
import WeightVisualiser from '../src/components/WeightVisualiser'

describe('WeightVisualiser', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(WeightVisualiser, {
      propsData: {
        weightList: [{
            name: 'Vue.js',
            weight: 50,
          },
          {
            name: 'jest.js',
            weight: 50,
          },
        ]
      }
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  describe('Props', () => {

    it('renders a list of projects', () => {
      expect(cmp.findAll('.weight-list .weight')).toHaveLength(cmp.vm.weightList.length)
    })

  })

  describe('Weight Style', () => {

    it('sets width to weight value', () => {
      let tech = cmp.vm.weightList[0]
      expect(cmp.find('.weight-list .weight').hasStyle('width', `${tech.weight}%`)).toBe(true)
    })

  })


})
