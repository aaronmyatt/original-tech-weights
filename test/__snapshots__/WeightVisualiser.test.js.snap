// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeightVisualiser has the expected html structure 1`] = `
<div
  class="weight-list progress"
  style="height: 20px;"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="25"
    class="weight progress-bar"
    role="progressbar"
    style="width: 50%; background-color: green;"
  >
    
    Vue.js: 50%
  
  </div>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow="25"
    class="weight progress-bar"
    role="progressbar"
    style="width: 50%; background-color: blue;"
  >
    
    jest.js: 50%
  
  </div>
</div>
`;
